text/0000-futures.md

```diff
@@ -154,31 +164,316 @@ can be dropped.
 
 ### Waking up
 
-Each task executor provides its own scheduling facilities, and hence needs to
-customize the way task wakeups are handled. Most of the time, you should use the
-`std::task::Wake` trait defining wakeup behavior:
+If a future can not be directly fulfilled during execution and returns `Pending`,
+it needs a way to later on inform the executor that it needs to get polled again
+to make progress.
+
+This functionality is provided through a set of `Waker` types.
+
+`Waker`s are objects which are passed as a parameter to the `Future::poll` call,
+and which can be stored by the implementation of those `Futures`s. Whenever a
+`Future` has the need to get polled again, it can use the `wake` method of the
+waker in order to inform the executor that the task which owns the `Future`
+should get scheduled and executed again.
+
+The RFC defines two concrete `Waker` types, with which implementors of `Futures`
+and asynchronous functions will interact: `Waker` and `LocalWaker`. Both of
+these types define a `wake(&self)` function which is used to schedule the task
+that is associated to the `Waker` again.
+
+The difference between those types is that `Waker` implements the `Send` and `Sync`
+marker traits, while `LocalWaker` doesn't. This means a `Waker` can be sent to
+another thread and stored there in order to wake up the associated task later on,
```

Comment:
```suggestion
another thread and stored there to wake up the associated task later on,
```

Reply:
I read both as "enabling an effect.."; As I said above (https://github.com/aturon/rfcs/pull/15#discussion_r239990861), "in order to" is syntactic salt... If you want to make any distinction in meaning clearer you can instead write:

> This means that a `Waker` can be sent to another thread and stored there so that later on, the associated task can be woken up. Meanwhile, a `LocalWaker` cannot be sent.